python/nvfuser_direct/pytorch_utils.py

```diff
@@ -35,8 +36,42 @@
     ] = DataType.Float4_e2m1fn_x2
 
 
+def python_scalar_to_nvfuser_dtype(a: Union[int, float, complex, bool]):
+    return _torch_dtype_to_nvfuser_dtype_map[type(a)]
+
+
 def torch_dtype_to_nvfuser_dtype(dtype: Union[torch.dtype, NumberTypeType]):
     """
     Translates from torch.dtype to nvFuser's DataType enum
     """
     return _torch_dtype_to_nvfuser_dtype_map[dtype]
+
+
+def retry_on_oom_or_skip_test(func):
+    """Decorator: upon torch.OutOfMemoryError clear the cache and retry test"""
+
+    @functools.wraps(func)
+    def retried_func(*args, **kwargs):
+        try:
+            output = func(*args, **kwargs)
+        except torch.OutOfMemoryError:
+            pass
+        else:
+            return output
+
+        # We have hit an OOM error, so clear the cache and retry
+        gc.collect()
+        torch.cuda.empty_cache()
+
+        try:
+            output = func(*args, **kwargs)
+        except torch.OutOfMemoryError as e:
+            # If we hit an OOM this time, then skip the test
+            import pytest
+
+            pytest.skip(f"Test failed due to OutOfMemoryError: {e}")
```

Comment:
OOC, does this pop up a lot during error testing?

Reply:
No idea.